### ISSUE TITLE
Improve itinerary description

### ISSUE THREAD

**Issue by perliedman** [perliedman/leaflet-routing-machine#9]
The text description of a route is sometimes hard to follow, and not very detailed. Look into what can be improved.

See for example: https://github.com/perliedman/leaflet-routing-machine/issues/6#issuecomment-38325060

**Comment by moimael** [comment]
Any progress on this ? I get not only undetailed but sometimes empty instructions, which look awful in the UI (it result in a empty list item).

**Comment by perliedman** [comment]
@moimael wow! Do you have an example (waypoint coords or similar) where this happens?

**Comment by moimael** [comment]
![capture d ecran de 2015-02-26 13 20 27](https://cloud.githubusercontent.com/assets/1479859/6391798/4fd36cbc-bdba-11e4-9ade-bc059dffeb51.png)

Just try to route from Paris to lille for example.
It happens almost on every route I try.

**Comment by TheMarex** [comment]
@moimael the frontend is not really to blame here. The main problems are:
- OSM has a lot of short unnamed streets (the 1s, 8s and 3s intervals)
- The instruction generation on the OSRM backend is pretty basic.

There is [some code in OSRM](https://github.com/Project-OSRM/osrm-backend/blob/master/descriptors/description_factory.hpp#L97) to generate better turn instructions, but currently it is too slow to be used in production:

**Comment by perliedman** [comment]
@TheMarex good to know. Do you know a relevant issue for OSRM-backend we could l link to here? https://github.com/Project-OSRM/osrm-backend/issues/685 seems like it could be related.

**Comment by moimael** [comment]
@TheMarex Thanks for the explaination ! Do you have any pointer to speed up OSRM code ?

**Comment by TheMarex** [comment]
@moimael to be honest its not a really priority at the moment. If you would like to take a step yourself, we are always happy to help new contributors.
